code/game/objects/effects/effect_system/smoke.dm

```diff
@@ -295,9 +295,9 @@
 	var/effect_amt = floor(6 + amount*6)
 
 	if(xeno_creature)
-		if(xeno_creature.interference < 4)
+		if(!HAS_TRAIT(xeno_creature, TRAIT_HIVEMIND_INTERFERENCE))
 			to_chat(xeno_creature, SPAN_XENOHIGHDANGER("Your awareness dims to a small area!"))
```

Comment:
It looks like this still gets spammed if you have immunity. So lets instead do it when its Initialized
```suggestion
```